flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/MongoDBSourceBuilder.java

```diff
@@ -0,0 +1,197 @@
+        checkArgument(pollAwaitTimeMillis > 0);
+        this.configFactory.pollAwaitTimeMillis(pollAwaitTimeMillis);
+        return this;
+    }
+
+    /**
+     * poll.max.batch.size
+     *
+     * <p>Maximum number of change stream documents to include in a single batch when polling for
+     * new data. This setting can be used to limit the amount of data buffered internally in the
+     * connector. Default: 1024
+     */
+    public MongoDBSourceBuilder<T> pollMaxBatchSize(int pollMaxBatchSize) {
+        this.configFactory.pollMaxBatchSize(pollMaxBatchSize);
+        return this;
+    }
+
+    /**
+     * copy.existing
+     *
+     * <p>Copy existing data from source collections and convert them to Change Stream events on
+     * their respective topics. Any changes to the data that occur during the copy process are
+     * applied once the copy is completed.
+     */
+    public MongoDBSourceBuilder<T> copyExisting(boolean copyExisting) {
+        if (copyExisting) {
+            this.configFactory.startupOptions(StartupOptions.initial());
+        } else {
+            this.configFactory.startupOptions(StartupOptions.latest());
+        }
+        return this;
+    }
+
+    /**
+     * heartbeat.interval.ms
+     *
+     * <p>The length of time in milliseconds between sending heartbeat messages. Heartbeat messages
+     * contain the post batch resume token and are sent when no source records have been published
+     * in the specified interval. This improves the resumability of the connector for low volume
+     * namespaces. Use 0 to disable.
+     */
+    public MongoDBSourceBuilder<T> heartbeatIntervalMillis(int heartbeatIntervalMillis) {
+        this.configFactory.heartbeatIntervalMillis(heartbeatIntervalMillis);
+        return this;
+    }
+
+    /**
+     * scan.incremental.snapshot.chunk.size.mb
+     *
+     * <p>The chunk size mb of incremental snapshot. Default: 64mb.
+     */
+    public MongoDBSourceBuilder<T> splitSizeMB(int splitSizeMB) {
+        this.configFactory.splitSizeMB(splitSizeMB);
+        return this;
+    }
+
+    /**
+     * The group size of split meta, if the meta size exceeds the group size, the meta will be
+     * divided into multiple groups.
+     */
+    public MongoDBSourceBuilder<T> splitMetaGroupSize(int splitMetaGroupSize) {
+        this.configFactory.splitMetaGroupSize(splitMetaGroupSize);
+        return this;
+    }
+
+    /**
+     * The deserializer used to convert from consumed {@link
+     * org.apache.kafka.connect.source.SourceRecord}.
+     */
+    public MongoDBSourceBuilder<T> deserializer(DebeziumDeserializationSchema<T> deserializer) {
+        this.deserializer = deserializer;
+        return this;
+    }
+
+    /**
+     * Build the {@link MongoDBSource}.
+     *
+     * @return a MongoDBParallelSource with the settings made for this builder.
+     */
+    public MongoDBSource<T> build() {
+        return new MongoDBSource<>(configFactory, checkNotNull(deserializer));
```

Comment:
What about having a sanity check here to validate if required options are present?

Reply:
Fixed.